File: src/frontend/types/query.ts
export interface QueryParams {
  connection_id: string;
  database: string;
  collection: string;
  query_text: string;
  timeout_ms?: number;
}

export interface QueryResult {
  documents: Record<string, unknown>[];
  total_count: number;
  execution_time_ms: number;
}

export interface QueryValidation {
  is_valid: boolean;
  error_message?: string;
}

export interface QueryHistoryEntry {
  id: string;
  connection_id: string;
  database: string;
  collection: string;
  query_text: string;
  execution_time_ms?: number;
  result_count?: number;
  is_favorite: boolean;
  created_at: string;
}

File: src-tauri/src/models/query.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryParams {
    pub connection_id: String,
    pub database: String,
    pub collection: String,
    pub query_text: String,
    pub timeout_ms: Option<u64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryResult {
    pub documents: Vec<serde_json::Value>,
    pub total_count: u64,
    pub execution_time_ms: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryValidation {
    pub is_valid: bool,
    pub error_message: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryHistoryEntry {
    pub id: String,
    pub connection_id: String,
    pub database: String,
    pub collection: String,
    pub query_text: String,
    pub execution_time_ms: Option<u64>,
    pub result_count: Option<u64>,
    pub is_favorite: bool,
    pub created_at: String,
}
